tta-server/app/logic/Logic.scala

```diff
@@ -1,32 +1,24 @@
 package logic
 
-import models.Action
-import models.ActionId
-import models.Bronze
-import models.BuildBronze
-import models.DerivedGameState
-import models.DerivedPlayerState
-import models.GameState
-import models.PlayerIndex
-import models.PlayerState
+import models._
 
 import scala.collection.immutable
 
 object Logic {
   
   def deriveGameState(gameState: GameState): DerivedGameState = {
-    val defaultCandidateActions: immutable.Seq[Action] = List(BuildBronze)
+    val defaultCandidateActionMap = gameState.activePlayerState.researchedBuildings.map (
+      building => {
+        ActionId("build" + building.prettyName) -> Building.generateBuildAction(building)
+      } //why does flatMap have an error here?
```

Comment:
`flatMap` expects your function to return a sequence-y thing but this function returns a single element, so `map` is more correct